content/testing.md

```diff
@@ -0,0 +1,23 @@
+# Testing Guide
+
+TODO: Explain why it is essential to have proper testing in our projects and for PRs to maintain tests up-to-date.
+
+## Unit Tests
+
+TODO: Introduce the subject with the example of how we made the Flow DPS mapper testable.
+TODO: Explain internal/external unit tests.
+TODO: Explain manual mocks.
+TODO: Explain pseudorandom generic values.
+TODO: Explain test parallelization.
+TODO: Explain how awesome the native `testing` package is.
+TODO: Explain subtest and structure of nested tests.
+TODO: Explain when to use subtests vs when to use table-driven tests.
+
+## Integration Tests
+
+TODO: Explain build tag usage to prevent integration tests from being run at all times.
+TODO: Explain how we write integration tests.
+
+## Benchmarks & Examples
+
+TODO: Explain how projects/packages that are meant to be used as libraries should ideally provide benchmarks and examples on top of proper documentation.
```

Comment:
Don't know if [this](https://go.dev/blog/examples) but could be a way to add examples with the go docs.

Reply:
Yep this is what we have in mind when we say examples :)

Reply:
Nice, I looked search a bit on flow-dps and didn't find any, that why I said it.

Reply:
Currently there's no package in Flow DPS that is meant to be used as a library by other projects, besides maybe the `storage` package that we recently made public I guess. So that's why we have no examples for now, but they'll be there should the need arise. For the case of the `storage` package I've thought about it and it's so straightforward that I'm not sure examples would be useful 🤔